src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets

```diff
@@ -70,6 +77,42 @@
                               ('$(RunAnalyzers)' == '' and '$(RunAnalyzersDuringBuild)' == 'false')">
       <_SkipAnalyzers>true</_SkipAnalyzers>
     </PropertyGroup>
+
+    <!-- PERF: For builds which are indirectly triggered inside Visual Studio from commands 
+               such as from 'Run Tests' or 'Start Debugging', we implicitly skip analyzers to speed up the build.
+               We only do so by default when 'TreatWarningsAsErrors' is not 'true'. For the scenario where
+			         'TreatWarningsAsErrors' is 'true', users can explicitly enable this functionality by setting
+			         'OptimizeIndirectlyTriggeredBuildInsideVisualStudio' to 'true'.
+    -->
+    <PropertyGroup Condition="'$(_SkipAnalyzers)' == '' and
+                              '$(IsIndirectlyTriggeredBuildInsideVisualStudio)' == 'true' and
+                              ('$(TreatWarningsAsErrors)' != 'true' or '$(OptimizeIndirectlyTriggeredBuildInsideVisualStudio)' == 'true')">
+      <_ImplicitlySkipAnalyzers>true</_ImplicitlySkipAnalyzers>
+      <_SkipAnalyzers>true</_SkipAnalyzers>
+    </PropertyGroup>
+
+    <!-- Display a message to inform the users about us implicitly skipping analyzers for speeding up indirect builds. -->
+    <Message Condition="'$(_ImplicitlySkipAnalyzers)' == 'true'"
+             Text="Skipping analyzers to speed up the build. You can execute 'Build' or 'Rebuild' command to run analyzers." Importance="High" />
+
+    <!-- Semaphore file to indicate the time stamp for last build with skipAnalyzers flag. -->
+    <PropertyGroup>
+      <LastBuildWithSkipAnalyzers>$(IntermediateOutputPath)$(MSBuildProjectFile).BuildWithSkipAnalyzers</LastBuildWithSkipAnalyzers>
```

Comment:
Have we verified that this logic works correctly with a multi-targeted build? Essentially does it create one of these files per target framework? Such that each TF is indpendently evaluated? 

I can't remember off hand if `$(IntermediateOutputPath)` is TF specific or not at this point in the build

Reply:
Thanks. Each TFM gets a separate intermediate path: `<%project_dir%>\obj\<%config%>\<%tfm%>\`. I just tested it works fine for multi-TFM project.